### PR TITLE
Adding Billing endpoints

### PR DESCRIPTION
**Pull request by jeromearz**

### CODE REVIEW

File: src/Billing/Enum/ConstantEnum.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+namespace ArrowSphere\PublicApiClient\Billing\Enum;
+
+use ArrowSphere\PublicApiClient\AbstractEnum;
+
+abstract class ConstantEnum extends AbstractEnum
+{
+    /**
+     * @var string periodFrom index
+     */
+    public const PERIODFROM = 'periodFrom';
```
Comment (pfeyssaguet) [inline src/Billing/Enum/ConstantEnum.php:12]: PERIOD_FROM would be more lisible

File: src/Billing/Enum/ConstantEnum.php [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    /**
+     * @var string periodTo index
+     */
+    public const PERIODTO = 'periodTo';
```
Comment (pfeyssaguet) [inline src/Billing/Enum/ConstantEnum.php:17]: Same here with PERIOD_TO

File: src/Billing/Enum/PreferenceTypeEnum.php [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+namespace ArrowSphere\PublicApiClient\Billing\Enum;
+
+use ArrowSphere\PublicApiClient\AbstractEnum;
+
+abstract class PreferenceTypeEnum extends AbstractEnum
+{
+    /**
+     * @var string GroupBy index
+     */
+    public const GROUPBY = 'GroupBy';
```
Comment (pfeyssaguet) [inline src/Billing/Enum/PreferenceTypeEnum.php:12]: Same here with GROUP_BY

File: src/Billing/Enum/PreferenceTypeEnum.php [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace ArrowSphere\PublicApiClient\Billing\Enum;
+
+use ArrowSphere\PublicApiClient\AbstractEnum;
+
+abstract class PreferenceTypeEnum extends AbstractEnum
```
Comment (pfeyssaguet) [inline src/Billing/Enum/PreferenceTypeEnum.php:7]: If this class is abstract, please name it AbstractPreferenceTypeEnum
See https://www.php-fig.org/bylaws/psr-naming-conventions/

File: src/Billing/Enum/ConstantEnum.php [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace ArrowSphere\PublicApiClient\Billing\Enum;
+
+use ArrowSphere\PublicApiClient\AbstractEnum;
+
+abstract class ConstantEnum extends AbstractEnum
```
Comment (pfeyssaguet) [inline src/Billing/Enum/ConstantEnum.php:7]: if this class is abstract, please name it AbstractConstantEnum
See https://www.php-fig.org/bylaws/psr-naming-conventions/

File: src/Billing/StatementsClient.php [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    /**
+     * Lists the statements.
+     * Returns an array (generator) of Statement.
+     *
+     * @param string $periodFrom YYYY-MM
+     * @param string $periodTo YYYY-MM
+     *
+     * @return Generator
```
Comment (pfeyssaguet) [inline src/Billing/StatementsClient.php:62]: `@return Generator|Statement[]`
If psalm whines about it, make it ignore this error with `make psalm-generate-baseline`

File: src/PublicApiClient.php [hunk 7]
```diff
@@ -119,4 +119,24 @@ public function getCustomersClient(): CustomersClient
             ->setUrl($this->url)
             ->setApiKey($this->apiKey);
     }
+
+    /**
+     * @return Billing\StatementsClient
+     */
+    public function getBillingStatementsClient(): Billing\StatementsClient
+    {
+        return (new Billing\StatementsClient($this->client))
```
Comment (pfeyssaguet) [inline src/PublicApiClient.php:128]: import the dependency with a use and remove the `Billing\` part

File: tests/Billing/Entities/PreferenceTest.php [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+namespace ArrowSphere\PublicApiClient\Tests\Billing\Entities;
+
+use ArrowSphere\PublicApiClient\Billing\Entities\Preference;
+use ArrowSphere\PublicApiClient\Exception\EntityValidationException;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Class PreferenceTest
+ */
+class PreferenceTest extends TestCase
```
Comment (pfeyssaguet) [inline tests/Billing/Entities/PreferenceTest.php:12]: There's a `AbstractEntityTest` class to standardize the testing of entities
But it's almost the same

File: tests/Billing/Entities/StatementBoundaryTest.php [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+namespace ArrowSphere\PublicApiClient\Tests\Billing\Entities;
+
+use ArrowSphere\PublicApiClient\Billing\Entities\StatementBoundary;
+use ArrowSphere\PublicApiClient\Exception\EntityValidationException;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Class StatementBoundaryTest
+ */
+class StatementBoundaryTest extends TestCase
```
Comment (pfeyssaguet) [inline tests/Billing/Entities/StatementBoundaryTest.php:12]: Same here with `AbstractEntityTest`

File: tests/Billing/Entities/StatementTest.php [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+namespace ArrowSphere\PublicApiClient\Tests\Billing\Entities;
+
+use ArrowSphere\PublicApiClient\Billing\Entities\Statement;
+use ArrowSphere\PublicApiClient\Exception\EntityValidationException;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Class StatementTest
+ */
+class StatementTest extends TestCase
```
Comment (pfeyssaguet) [inline tests/Billing/Entities/StatementTest.php:12]: Same here with `AbstractEntityTest`

**Review by pfeyssaguet — CHANGES_REQUESTED** [review]
Please add some documentation (see the doc directory and the md files)
Please also add some info in the changelog to tell about your changes

File: src/PublicApiClient.php (hunk 7 above)
Comment (A84126) [reply src/PublicApiClient.php:128]: Preferences and Statements are words common enough to be misleading without the namespace
Is it okay to leave it this way ? or do you want to remove the namespace just for stylistic issues ?

**Pull request merged**